### PR TITLE
358 restart session on time limit

### PR DESCRIPTION
**Pull request by aaronyoung-tw**
## Summary

update webchax component and helpers to refresh the page after 30 min of 
  inactivity with the chat bot 
- *(If bug, how to reproduce)*
  microsoft PVA's sessions only last 30 minutes so we need to start a new one
- *(What is the solution, why is this the solution)*
- reload the page at the 30 min mark, otherwise there are gibberish (to the 
  user) questions
- *(Which team do you work for, does your team own the maintainence of this component?)*
- chatbot team

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#0000
- *Link to ticket created in va.gov-team repo*
- [Ticket](https://github.com/department-of-veterans-affairs/va-virtual-agent/issues/358)

## Testing done

- *Describe what the old behavior was prior to the change*
we see weird programming questions as the chat session restarts after 30 min
- *Describe the steps required to verify your changes are working as expected*
wait 30 minutes after last activity with the bot and experience a page reload
- *Describe the tests completed and the results*
written automated tests that pass

## Screenshots
_Note: This field is mandatory for component work and UI changes (non-component work should NOT have screenshots)._
N/A

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
Virtual agent
## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
    - N/A
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
    - N/A
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
    - N/A 
- [ ]  I added a screenshot of the developed feature
    - N/A

### CODE REVIEW

**Review by joehall-tw — APPROVED** [review]
LGTM. I didn't pull and test, but the flow looks fine.

**Comment by khathuynh-tw** [comment]
Looks good for the 30 min idleness scenario. However, we will also need refresh handling for 60 min scenario (regardless of idleness).

Re: unit testing, can we make use of mocks to mock the timer timing out?

File: src/applications/virtual-agent/tests/Chatbox.unit.spec.js [hunk 1]
```diff
@@ -923,6 +923,74 @@ describe('App', () => {
...
+      renderInReduxProvider(
+        <Chatbox {...defaultProps} />,
+        unacknowledgedUserStore,
+      );
+      const incomingActivityEvent1 = new Event('bot-incoming-activity');
+      const testNumber = 7;
+      incomingActivityEvent1.data = testNumber;
+      window.dispatchEvent(incomingActivityEvent1);
+      sandbox.clock.tick(1);
+      expect(fakeClearTimeout.called).to.be.false;
```
Comment (khathuynh-tw) [inline src/applications/virtual-agent/tests/Chatbox.unit.spec.js:986]: Curious about this second test case. Want to understand why the significance of the window event `data` property and why we expect the timer to not be called at first.

**Review by khathuynh-tw — COMMENTED** [review]
LGTM. Test coverage is there, just have a question about one of the cases

**Comment by aaronyoung-tw** [comment]
This is the way the chatbot.jsx receives a timeoutID - we needed to
validate that it was being used appropriately

**Comment by adamwhitlock1** [comment]
@aaronyoung-tw I'm seeing test failures on main for the Chatbot unit tests

https://github.com/department-of-veterans-affairs/vets-website/actions/runs/4144423936/jobs/7167454280

Any idea why that might be occuring?

**Comment by aaronyoung-tw** [comment]
Thats very interesting as it passed on the branch - i don't know why it would be failing but i am going to run it again

**Comment by aaronyoung-tw** [comment]
It reran fine on the second run

**Pull request merged**